src/wallet/wallet2.cpp

```diff
@@ -3466,72 +3536,69 @@ std::string wallet2::make_multisig(const epee::wipeable_string &password,
   CHECK_AND_ASSERT_THROW_MES(!view_keys.empty(), "empty view keys");
   CHECK_AND_ASSERT_THROW_MES(view_keys.size() == spend_keys.size(), "Mismatched view/spend key sizes");
   CHECK_AND_ASSERT_THROW_MES(threshold > 1 && threshold <= spend_keys.size() + 1, "Invalid threshold");
-  CHECK_AND_ASSERT_THROW_MES(threshold == spend_keys.size() || threshold == spend_keys.size() + 1, "Unsupported threshold case");
 
   std::string extra_multisig_info;
-  crypto::hash hash;
-
-  clear();
-
-  MINFO("Creating spend key...");
   std::vector<crypto::secret_key> multisig_keys;
-  rct::key spend_pkey, spend_skey;
+  rct::key spend_pkey = rct::identity();
+  rct::key spend_skey;
+  std::vector<crypto::public_key> multisig_signers;
+
   if (threshold == spend_keys.size() + 1)
   {
+    // N / N case
+    MINFO("Creating spend key...");
     cryptonote::generate_multisig_N_N(get_account().get_keys(), spend_keys, multisig_keys, spend_skey, spend_pkey);
+
+    multisig_signers = spend_keys;
+
+    crypto::public_key signer;
+    CHECK_AND_ASSERT_THROW_MES(crypto::secret_key_to_public_key(get_account().get_keys().m_spend_secret_key, signer), "Failed to generate signer public key");
```

Comment:
using this function here would throw because at the moment is not in multisig state

Reply:
That sounds either unlikely, or a hint of a bug somewhere.

Reply:
Why does it sound unlikely? `get_multisig_signer_public_key` meant to be used when wallet creation is already finished and in `multisig` state.

At the moment we can calculate secret spend key for each participant only in case of classic N/N or N-1/N schemes and get permanent signer public key. In other cases secret spend can be calculated  only before last round.

If you want this key to be derived with `get_multisig_signer_public_key` I can remove the check inside of it. Btw, in previous version of code this function hasn't been used either. 

I can't see buggy behavior here for now. If you see better solution please let me know

Reply:
okay, will be done